src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -158,8 +164,44 @@ def _populate_core_stats(self, area):
                 for trade in area.current_market.trades:
                     core_stats_dict['trades'].append(trade.serializable_dict())
 
+            if isinstance(area.strategy, PVStrategy) or \
+                    isinstance(area.strategy, PVUserProfileStrategy) or \
+                    isinstance(area.strategy, PVPredefinedStrategy):
+                core_stats_dict['pv_production_kWh'] = \
+                    area.strategy.energy_production_forecast_kWh.get(self.current_market_datetime,
+                                                                     0)
+                if area.parent.current_market is not None:
+                    for t in area.strategy.trades[area.parent.current_market]:
+                        core_stats_dict['trades'].append(t.serializable_dict())
+
+            elif isinstance(area.strategy, StorageStrategy):
+                core_stats_dict['soc_history_%'] = \
+                    area.strategy.state.charge_history.get(self.current_market_datetime, 0)
+                if area.parent.current_market is not None:
+                    for t in area.strategy.trades[area.parent.current_market]:
+                        core_stats_dict['trades'].append(t.serializable_dict())
+
+            elif isinstance(area.strategy, LoadHoursStrategy):
+                core_stats_dict['load_profile_kWh'] = \
+                    area.strategy.state.desired_energy_Wh.get(self.current_market_datetime, 0)
+                if area.parent.current_market is not None:
+                    for t in area.strategy.trades[area.parent.current_market]:
+                        core_stats_dict['trades'].append(t.serializable_dict())
+
+            elif type(area.strategy) == FinitePowerPlant:
+                core_stats_dict['production_kWh'] = area.strategy.energy_per_slot_kWh
+                if area.parent.current_market is not None:
+                    for t in area.strategy.trades[area.parent.current_market]:
+                        core_stats_dict['trades'].append(t.serializable_dict())
+
+            elif type(area.strategy) == InfiniteBusStrategy:
+                core_stats_dict['production_kWh'] = area.strategy.energy_per_slot_kWh
+                if area.parent.current_market is not None:
+                    for t in area.strategy.trades[area.parent.current_market]:
+                        core_stats_dict['trades'].append(t.serializable_dict())
+
             self.flattened_area_core_stats_dict[area.uuid] = {}
```

Comment:
Can you tell me why you decided to move this into the endpoint buffer? AFAIR, this is device statistics code and i dont see it being useful for other statistics. 
Also, Is there a reason why you are writing the `production_kWh`, `load_profile_kWh` etc directly to the `self.flattened_area_core_stats_dict[area.uuid][self.current_market]` instead of having another subdict called "device_statitics" or so?

Reply:
Yes, that is the whole essence of `flattened_area_core_stats_dict ` to have all kinds of raw data being needed for results at one place, for example `load_profile_kWh ` is not just needed for `device_statistics` but also for `unmatched_loads`, so rather than having the same stats/data being duplicated at multiple places is to have in one common pleace

Reply:
ok, makes sense. Thanks!